handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/GetAccount.scala

```diff
@@ -80,35 +82,26 @@ object GetAccount {
     )
 
     given JsonDecoder[BasicInfo] = DeriveJsonDecoder.gen[BasicInfoWire].map {
-      case BasicInfoWire(defaultPaymentMethod, identityId, sfContactId__c, balance, currency) =>
+      case BasicInfoWire(id, defaultPaymentMethod, identityId, sfContactId__c, balance, currency) =>
         identityId match {
-          case None =>
+          case Some(someIdentityId) if someIdentityId == "" =>
+            BasicInfo(
+              id = id,
+              defaultPaymentMethod = defaultPaymentMethod,
+              IdentityId__c = None,
+              sfContactId__c = sfContactId__c,
+              balance = (balance.toDouble * 100).toInt,
+              currency = currencyCodetoObject(currency),
+            )
+          case _ =>
             BasicInfo(
-              defaultPaymentMethod,
-              None,
-              sfContactId__c,
-              (balance.toDouble * 100).toInt,
-              currencyCodetoObject(currency),
+              id = id,
+              defaultPaymentMethod = defaultPaymentMethod,
+              IdentityId__c = identityId,
+              sfContactId__c = sfContactId__c,
+              balance = (balance.toDouble * 100).toInt,
+              currency = currencyCodetoObject(currency),
             )
-          case Some(id) =>
-            id match {
-              case "" =>
-                BasicInfo(
-                  defaultPaymentMethod,
-                  None,
-                  sfContactId__c,
-                  (balance.toDouble * 100).toInt,
-                  currencyCodetoObject(currency),
-                )
-              case _ =>
-                BasicInfo(
-                  defaultPaymentMethod,
-                  Some(id),
-                  sfContactId__c,
-                  (balance.toDouble * 100).toInt,
-                  currencyCodetoObject(currency),
-                )
-            }
         }
```

Comment:
maybe there used to be a difference but not any more?
```suggestion
        BasicInfo(
          id = id,
          defaultPaymentMethod = defaultPaymentMethod,
          IdentityId__c = identityId.filter(_ != ""),
          sfContactId__c = sfContactId__c,
          balance = (balance.toDouble * 100).toInt,
          currency = currencyCodetoObject(currency),
        )
```